### PR TITLE
HPCC-31671 Add resolved "CommandMessage" field to GetDFUWorkunits result.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Tested locally

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/services/ws_fs/ws_fsService.cpp [hunk 1]
```diff
@@ -1113,9 +1113,12 @@ bool CFileSprayEx::onGetDFUWorkunits(IEspContext &context, IEspGetDFUWorkunits &
...
             Owned<IEspDFUWorkunit> resultWU = createDFUWorkunit("", "");
             resultWU->setID(wu->queryId());
-            StringBuffer jobname, user, cluster;
+            StringBuffer jobname, user, cluster, cmdStr;
             resultWU->setJobName(wu->getJobName(jobname).str());
-            resultWU->setCommand(wu->getCommand());
+            DFUcmd command = wu->getCommand();
+            resultWU->setCommand(command);
+            encodeDFUcommand(command, cmdStr);
+            resultWU->setCommandMessage(cmdStr.str());
```
Comment (jakesmith) [inline esp/services/ws_fs/ws_fsService.cpp:1121]: >     [min_ver("1.03")] string CommandMessage;

This code should be wrapped in a version check (>= 1.03):
```
            if (version >= 1.03)
            {
                encodeDFUcommand(command, cmdStr);
                resultWU->setCommandMessage(cmdStr.str());
            }
```

**Review by jakesmith — COMMENTED** [review]
@AttilaVamos - one minor issue

File: esp/services/ws_fs/ws_fsService.cpp [hunk 2]
```diff
@@ -1113,12 +1113,16 @@ bool CFileSprayEx::onGetDFUWorkunits(IEspContext &context, IEspGetDFUWorkunits &
...
             Owned<IEspDFUWorkunit> resultWU = createDFUWorkunit("", "");
             resultWU->setID(wu->queryId());
-            StringBuffer jobname, user, cluster, cmdStr;
+            StringBuffer jobname, user, cluster;
             resultWU->setJobName(wu->getJobName(jobname).str());
             DFUcmd command = wu->getCommand();
             resultWU->setCommand(command);
-            encodeDFUcommand(command, cmdStr);
-            resultWU->setCommandMessage(cmdStr.str());
+            if ( version >=1.03 )
```
Comment (jakesmith) [inline esp/services/ws_fs/ws_fsService.cpp:1120]: trivial/style: non-standard spacing.

**Review by jakesmith — COMMENTED** [review]
@AttilaVamos - it's fine, but see comment re. unusual spacing

**Review by jakesmith — APPROVED** [review]
@AttilaVamos - looks good.

**Comment by jakesmith** [comment]
@AttilaVamos - please squash

**Comment by AttilaVamos** [comment]
@jakesmith squashed.
@ghalliday please merge.

**Pull request merged**